lib/client.js

```diff
@@ -633,19 +634,61 @@ MatrixClient.prototype.isEventSenderVerified = function(event) {
 };
 
 /**
+ * Register a listener for m.room.encryption events which will enable encryption
+ * for a room.
+ *
+ * @param {MatrixClient} client
+ */
+function setupCryptoEventHandler(client) {
+    client.on("event", function(event) {
+        if (event.getType() != "m.room.encryption") {
```

Comment:
Should this check to see if it's a state event too? Otherwise could it get confused if you send a non-state m.room.encryption which enables encryption but then you connect with another client which doesn't enable crypto because the event has scrolled off the history?